apps/server/src/services/ElasticSearchClient.ts

```diff
@@ -34,38 +35,56 @@ export async function initialize() {
     index: 'institutions'
   })
   if (!elasticSearchLoaded) {
-    await reIndexElasticSearch()
+    await indexElasticSearch()
   } else {
     info('ElasticSearch already indexed')
   }
 }
 
-export async function reIndexElasticSearch() {
-  try {
-    await ElasticsearchClient.indices.delete({
-      index: 'institutions'
-    })
-  } catch {
-    info('Elasticsearch "institutions" index did not exist')
-  }
-  info('Elasticsearch indexing institutions')
-  const dataFilePath = getInstitutionFilePath()
-  const rawData = readFileSync(dataFilePath)
-  const jsonData = JSON.parse(rawData.toString())
+export async function indexElasticSearch() {
+  const institutionData = await getInstitutions()
+  const insIds = institutionData.map((ins: CachedInstitution) => ins.ucp_id)
+
+  await overwriteSet(INSTITUTION_CURRENT_LIST_IDS, insIds)
 
   await ElasticsearchClient.indices.create({ index: 'institutions' })
 
-  const indexPromises = jsonData.map(async (institution: { ucp_id: any }) => {
-    return await ElasticsearchClient.index({
-      index: 'institutions',
-      id: institution.ucp_id,
-      document: institution
-    })
-  })
+  const indexPromises = institutionData.map(
+    async (institution: { ucp_id: any }) => {
+      return await ElasticsearchClient.index({
+        index: 'institutions',
+        id: institution.ucp_id,
+        document: institution
+      })
+    }
+  )
 
   await Promise.all(indexPromises)
 }
 
+async function getInstitutions(): Promise<CachedInstitution[]> {
+  const response = await fetchInstitutions()
+  if (response?.ok) {
+    info('Elasticsearch indexing from server list')
+    const newInstitutions = await response.json()
```

Comment:
Need a try catch for response.json(), and I'd do newInstitutions?.length so it doesn't blow up if it's empty